sphinx_gallery/gen_gallery.py

```diff
@@ -143,28 +160,57 @@ def touch_empty_backreferences(app, what, name, obj, options, lines):
         open(examples_path, 'w').close()
 
 
-gallery_conf = {
-    'filename_pattern': re.escape(os.sep) + 'plot',
-    'examples_dirs': '../examples',
-    'gallery_dirs': 'auto_examples',
-    'mod_example_dir': os.path.join('modules', 'generated'),
-    'doc_module': (),
-    'reference_url': {},
-}
+def sumarize_failing_examples(app, exception):
+    """Embed hyperlinks to documentation into example code"""
+    if exception is not None:
+        return
+
+    gallery_conf = app.config.sphinx_gallery_conf
+    failing_examples = set(gallery_conf['failing_examples'])
+    expected_failing_examples = set(gallery_conf['expected_failing_examples'])
+
+    still_failing_examples = failing_examples.intersection(
+        expected_failing_examples)
+    expected_fail_msg = []
+    if still_failing_examples:
+        expected_fail_msg.append("Known failing examples, still failing:")
+        for fail_example in still_failing_examples:
+            expected_fail_msg.append(fail_example + ' failed leaving traceback:\n' +
+                                     gallery_conf['failing_examples'][fail_example] + '\n')
+        print("\n".join(expected_fail_msg))
+
+    unexpected_ex_fails = failing_examples.difference(
+        expected_failing_examples)
+    fail_msgs = []
+    if unexpected_ex_fails:
+        fail_msgs.append("Unexpected failing examples:")
+        for fail_example in unexpected_ex_fails:
+            fail_msgs.append(fail_example + ' failed leaving traceback:\n' +
+                             gallery_conf['failing_examples'][fail_example] + '\n')
+
+    unexpected_ex_pass = expected_failing_examples.difference(failing_examples)
+    if unexpected_ex_pass:
+        fail_msgs.append("Examples that are not failing anymore:\n" +
+                         "\n".join(unexpected_ex_pass))
+
+    if fail_msgs:
+        raise ValueError("Here is a summary of the problems encountered when"
+                         "running the examples\n\n" + "\n".join(fail_msgs) + "-" * 79)
 
 
 def setup(app):
     """Setup sphinx-gallery sphinx extension"""
     app.add_config_value('plot_gallery', True, 'html')
     app.add_config_value('abort_on_example_error', False, 'html')
-    app.add_config_value('sphinx_gallery_conf', gallery_conf, 'html')
+    app.add_config_value('sphinx_gallery_conf', DEFAULT_GALLERY_CONF, 'html')
```

Comment:
Something like this seems to do what #122 wanted. Feel free to refactor it the way you like:

```diff
diff --git a/sphinx_gallery/gen_gallery.py b/sphinx_gallery/gen_gallery.py
index 32bdf3a..4147e75 100644
--- a/sphinx_gallery/gen_gallery.py
+++ b/sphinx_gallery/gen_gallery.py
@@ -198,11 +198,20 @@ def sumarize_failing_examples(app, exception):
                          "running the examples\n\n" + "\n".join(fail_msgs) + "-" * 79)
 
 
+def get_default_plot_gallery(conf):
+    return conf['sphinx_gallery_conf'].get('plot_gallery', True)
+
+
+def get_default_abort_on_example(conf):
+    return conf['sphinx_gallery_conf'].get('abort_on_example_error', False)
+
+
 def setup(app):
     """Setup sphinx-gallery sphinx extension"""
-    app.add_config_value('plot_gallery', True, 'html')
-    app.add_config_value('abort_on_example_error', False, 'html')
     app.add_config_value('sphinx_gallery_conf', DEFAULT_GALLERY_CONF, 'html')
+    app.add_config_value('plot_gallery', get_default_plot_gallery, 'html')
+    app.add_config_value('abort_on_example_error', get_default_abort_on_example, 'html')
+
     app.add_stylesheet('gallery.css')
 
     if 'sphinx.ext.autodoc' in app._extensions:
```

Basically the order of priority is (from lower to higher priority):
- DEFAULT_GALLERY_CONF
- sphinx_gallery_conf (in conf.py)
- override provided to sphinx-build through the `-D` flag